src/Aspire.Hosting/LaunchProfileExtensions.cs

```diff
@@ -105,31 +106,30 @@ private static bool TrySelectLaunchProfileByOrder(ProjectResource projectResourc
         return true;
     }
 
-    private static bool TrySelectLaunchProfileFromEnvironment(ProjectResource projectResource, [NotNullWhen(true)] out string? launchProfileName)
+    private static bool TrySelectLaunchProfileFromDefaultAnnotation(ProjectResource projectResource, [NotNullWhen(true)] out string? launchProfileName)
     {
-        var launchProfileEnvironmentVariable = Environment.GetEnvironmentVariable("DOTNET_LAUNCH_PROFILE");
```

Comment:
Moved to `WithProjectDefaults` via `DefaultLaunchProfileAnnotation`. See https://github.com/dotnet/aspire/pull/4287/files#diff-9e18fbbbb697e4ee2da32654325b13f2d6b8b8659e5a34747a63c8a81781aaedR227